### DISCUSSION TITLE
How to log user's IP during sign in attempt?

### DISCUSSION THREAD

**Discussion by nandox5** [nextauthjs/next-auth#5693] [Help]
I need to be able to log the ip where a user signs in from. Is there a way to access the req from nextjs during the signin() callback.

or is there a better way of achieving this?

**Comment by ThangHuuVu** [comment]
You can use https://next-auth.js.org/configuration/initialization#advanced-initialization

**Reply by nandox5** [reply]
This does not work, as the session doesn't seem to be accessible in signIn callback. Any other ideas?

**Reply by nandox5** [reply]
Can you provide an example of how to accomplish based on my question? need to get ip during signin and log it.

thanks in advanced

**Comment by nandox5** [comment]
bumping this question.. any help? need to really access the req during signin.  while eliminating code duplication
